logger/logging_config.py

```diff
@@ -0,0 +1,18 @@
+import json
+from logging.config import dictConfig  # type: ignore
+
+done_setup = False
```

Comment:
You can use a memoization decorator to remove the need to use a variable as a global sentry, like this:

class memoize:
  def __init__(self, function):
    self.function = function
    self.memoized = {}
  def __call__(self, *args):
    try:
      return self.memoized[args]
    except KeyError:
      self.memoized[args] = self.function(*args)
      return self.memoized[args]

Or, other examples: 
http://code.activestate.com/recipes/425445-once-decorator/
http://pydev.blogspot.co.uk/2012/12/python-tricks-making-sure-function-is.html